### ISSUE TITLE
Issue 8: Revise the manuscript around evaluation shortcuts and topology contribution

### ISSUE THREAD

**Issue by Mati86** [AI-Security-Internships-2026/12-graph-based-cyber-physical-risk#29]
# Issue 8: Revise the manuscript around evaluation shortcuts and topology contribution

## Milestone: M4 — Robustness, Statistics and Paper Results

---

## Problem Statement

Rewrite the current manuscript so that the paper is no longer organized mainly as a chronological internship story.

The revised paper should make one focused scientific argument:

> GNN-based ICS intrusion-detection performance can be substantially affected by evaluation protocol, identity/topology shortcuts, and temporal distribution shift; therefore graph contribution must be validated through leakage-safe split auditing and controlled topology/content counterfactuals.

---

### 1. Final Research Questions

Use approximately the following RQs.

#### RQ1 — Evaluation protocol

**How strongly do common train/test partitioning strategies affect reported GNN-based ICS intrusion-detection performance?**

Evidence:

- split audit,
- Track A vs Track B,
- strict temporal evaluation,
- BATADAL purged temporal evaluation.

#### RQ2 — Graph contribution

**Under leakage-safe evaluation, how much predictive value comes from genuine graph topology compared with packet/flow content and shortcut features?**

Evidence:

- MLP/tree vs GNN,
- content-only,
- topology-only,
- random topology,
- degree-preserving rewiring,
- no-message-passing ablation.

#### RQ3 — Robustness

**Do the observed shortcut and generalization effects remain consistent across datasets, time, attack types, model families, and random seeds?**

Evidence:

- multi-seed results,
- per-attack analysis,
- BATADAL/SCADANet,
- explanation/intervention study,
- operational metrics.

### 2. Scope Correction

The paper should **not** imply that the work has completed full:

- attack-path prediction,
- propagation-likelihood prediction,
- cyber-to-physical consequence modeling,
- physical impact regression,

unless these are actually implemented and experimentally evaluated.

The ICS knowledge graph may remain as:

- supporting system context,
- exploratory attack-path representation,
- supplementary implementation evidence,

but should not dominate the claimed contribution unless an experiment shows it materially improves the method.

### 3. Recommended Title Direction

Use a title closer to the actual contribution.

Examples:

```text
Evaluation-Safe Graph Neural Networks for ICS Intrusion Detection:
Auditing Split Shortcuts, Topology Dependence, and Temporal Generalization
```

or:

```text
When Graph Neural Network Accuracy Misleads:
Evaluation Shortcuts and Topology Counterfactuals in ICS/SCADA Intrusion Detection
```

Final wording can be adjusted to the target journal.

#### 1. Introduction

Clearly state:

- why graph-based ICS intrusion detection is attractive,
- why apparently strong GNN metrics can be misleading,
- why graph topology must be shown to matter,
- why temporal evaluation matters.

End with 3–4 concise contributions.

#### 2. Related Work

Organize by research problem rather than dataset chronology:

- GNN-based intrusion detection,
- ICS/SCADA graph detection,
- evaluation leakage and generalization,
- shortcut learning,
- explainability and topology dependence,
- temporal/spatio-temporal ICS detection.

Clearly distinguish the revised paper from work that proposes new GNN architectures.

#### 3. Methodology

Include:

- datasets,
- graph construction,
- feature construction,
- split protocols,
- split-audit framework,
- models/baselines,
- topology/content counterfactuals,
- strict temporal protocol,
- host-normalization mitigation,
- metrics and statistics.

#### 4. Experimental Setup

Report:

- hardware,
- software/library versions,
- seeds,
- hyperparameter search space,
- threshold calibration procedure,
- data cleaning,
- exact split sizes.

#### 5. Results

Organize by RQ.

##### 5.1 RQ1 — Evaluation protocol

Use audit and temporal results.

##### 5.2 RQ2 — Does topology actually help?

Use baselines + ablations.

##### 5.3 RQ3 — Robustness and operational implications

Use seeds, statistics, per-attack behavior, host mitigation, explainability/intervention.

#### 6. Discussion

Discuss:

- what inflated metrics,
- when topology helps,
- when content dominates,
- attack-specific failure modes,
- limits of benchmark datasets,
- implications for GNN-IDS evaluation practice.

#### 7. Threats to Validity / Limitations

Explicitly discuss:

- dataset-specific host structure,
- simulated vs real ICS data,
- attack-event independence,
- limited number of datasets,
- explainability limitations,
- lack of full physical-process impact modeling.

#### 8. Conclusion

Answer each RQ directly.

### 5. Required Main-Paper Tables

The final manuscript should contain a compact set of high-value tables.

#### Table 1 — Dataset and graph summary

| Dataset | Unit | Samples/windows | Graph nodes | Graph edges | Attack classes | Main protocol |
|---|---|---:|---:|---:|---:|---|

#### Table 2 — Split audit

Use outputs from Issue #2.

| Dataset | Protocol | Attack coverage | Host overlap | Pair overlap | Temporal overlap | Main warning |
|---|---|---:|---:|---:|---|---|

#### Table 3 — Fair model baselines

Use Issue #3.

| Protocol | Model | Precision | Recall | F1 | AUPRC | FPR |
|---|---|---:|---:|---:|---:|---:|

#### Table 4 — Topology/content ablation

Use Issue #4.

| Condition | Content | True topology | F1 | AUPRC | FPR | ΔF1 |
|---|---|---|---:|---:|---:|---:|

#### Table 5 — Strict temporal generalization

Use Issue #5.

| Dataset | Model | Non-temporal F1 | Strict temporal F1 | ΔF1 | Temporal recall | FPR |
|---|---|---:|---:|---:|---:|---:|

#### Table 6 — Mitigation and robustness

Use Issues #6–#7.

| Condition | F1 mean ± std | 95% CI | FPR | Key finding |
|---|---:|---:|---:|---|

Additional detailed tables should move to supplementary material.

#### Figure 1 — Proposed evaluation-audit framework

Suggested flow:

```text
ICS/SCADA dataset
      ↓
Graph construction
      ↓
Split protocol
      ↓
Leakage / shortcut audit
      ↓
Model training
      ↓
Topology/content counterfactuals
      ↓
Strict temporal evaluation
      ↓
Robustness + operational metrics
```

#### Figure 2 — Why split choice changes the task

Show attack-type coverage and identity overlap across:

- Track A,
- Track B,
- temporal.

This should visually expose why a high aggregate result can be misleading.

#### Figure 3 — Does true topology matter?

Use the key Issue #4 result:

- content only,
- topology only,
- full GraphSAGE,
- random topology,
- degree-preserved topology.

This is likely the **most important figure in the paper**.

#### Figure 4 — Strict temporal degradation by attack type

Show static/non-temporal vs strict temporal recall per SCADANet attack class.

#### Figure 5 — Host-specific false positives and mitigation

Show:

- false-positive concentration by host,
- before/after host normalization/calibration.

#### Figure 6 — Robustness across seeds

Show principal model/ablation performance distributions with confidence intervals.

Keep approximately 5–6 main figures. Move extra diagnostics to supplementary material.

### 7. Required Headline Results to Report

The paper should explicitly report, after all new experiments:

#### Result R1 — Split inflation

Quantify how much headline performance changes when evaluation moves from a shortcut-prone/limited-coverage protocol to a broader or stricter protocol.

#### Result R2 — Genuine topology contribution

Quantify:

```text
GraphSAGE(full true graph) - content-only
GraphSAGE(full true graph) - random topology
GraphSAGE(full true graph) - degree-preserved rewiring
```

This result is central regardless of whether the differences are large or small.

#### Result R3 — Temporal generalization gap

Report strict causal temporal performance, not the old approximate temporal result as the main claim.

#### Result R4 — Host-shortcut mitigation

Quantify whether host normalization reduces:

- normal FPR,
- concentration of false positives,

without causing unacceptable recall loss.

#### Result R5 — Robustness

Report seed-averaged estimates and 95% confidence intervals rather than best-run metrics.

### 8. Result Interpretation Rules

Do not write:

> "GraphSAGE is superior"

unless the fair baseline and topology-counterfactual experiments support it.

Do not write:

> "Topology causes the prediction"

based only on explainer mask magnitudes.

Do not write:

> "Temporal performance drops by X because of feature drift"

unless the new experiments isolate that explanation.

Do write precise conclusions such as:

> "Under strict chronological evaluation, recall for attack class X falls from A to B, while the same model remains stable for attack class Y."

### 9. Repository Cleanup Before Submission

Before final manuscript submission:

- update README to identify the experiment entry points,
- ensure final result-generation scripts are committed,
- ensure dependencies/environment are documented,
- remove or mark obsolete provisional result files,
- do not delete old history needed for traceability,
- ensure paper values come from final result artifacts,
- add a results-to-paper mapping file if useful.

Suggested:

```text
docs/paper-result-map.md
```

Example:

```text
Table 3 row GraphSAGE Track B
→ experiments/results/q1/baselines/scadanet_trackb_baselines.csv

Figure 3
→ experiments/results/q1/ablations/scadanet_trackb_topology_ablation.csv
```

---

## Final Acceptance Criteria

- [ ] Paper is structured around RQ1–RQ3.
- [ ] Main claims match actual implemented experiments.
- [ ] Unsupported attack-path/physical-impact claims are removed or clearly labeled future work.
- [ ] All tables are generated from committed final results.
- [ ] All figures are generated from committed final results.
- [ ] Fair non-GNN/GNN baselines are included.
- [ ] Topology/content counterfactual experiments are central to the paper.
- [ ] Strict leakage-free temporal results replace approximate temporal results as the main evidence.
- [ ] Multi-seed/statistical uncertainty is reported.
- [ ] Negative findings are preserved.
- [ ] Limitations/threats to validity are explicit.
- [ ] README and reproduction instructions match the final code.
- [ ] Manuscript is ready for supervisor review before target-journal formatting.

---

## Do Not Do In This Issue

Do not:

- add unrelated methods merely to increase page count,
- claim novelty from GraphSAGE itself,
- hide negative ablation results,
- retain outdated metrics because they look better,
- overstate the knowledge-graph component,
- submit until every headline number can be traced to a committed result artifact.
